### PR TITLE
feat(cb2-6440): create new templates for new vehicle types

### PR DESCRIPTION
**Pull request by naathanbrown**
## View a light vehicle Tech Record

Create new vehicle types and templates

PLEASE IGNORE VALIDATION RULES ETC, THIS WILL BE COVERED IN AMEND TICKET
[https://dvsa.atlassian.net/browse/CB2-6640](6640)

## Checklist

- [ ] Branch is rebased against the latest develop/common
- [ ] Necessary `id` required prepended with `"test-"` have been checked with automation testers and added
- [ ] Code and UI has been tested manually after the additional changes
- [ ] PR title includes the JIRA ticket number
- [ ] Squashed commits contain the JIRA ticket number
- [ ] Link to the PR added to the repo
- [ ] Delete branch after merge

### CODE REVIEW

**Comment by tahayazici** [comment]
Need to fix build and unit tests.

File: src/app/api/test-types/model/testType.ts [hunk 1]
```diff
@@ -80,14 +80,15 @@ export interface TestType {
     forVehicleWheels?: Array<number>;
 }
 export namespace TestType {
-    export type ForVehicleTypeEnum = 'psv' | 'hgv' | 'trl' | 'car' | 'lgv' | 'motorcycle';
+    export type ForVehicleTypeEnum = 'psv' | 'hgv' | 'trl' | 'car' | 'lgv' | 'motorcycle' | 'small trailer';
```
Comment (leifkemp-bjss) [inline src/app/api/test-types/model/testType.ts:83]: I think we're classifying trailers as small trailers by their EU vehicle category, and not a new vehicle type

Comment (naathanbrown) [reply src/app/api/test-types/model/testType.ts:83]: Again that isn't what's on the documentation, but I will look into this.

**Review by leifkemp-bjss — APPROVED** [review]

File: src/app/forms/custom-sections/dimensions/dimensions.component.ts [hunk 2]
```diff
@@ -24,7 +24,7 @@ export class DimensionsComponent implements OnInit, OnChanges, OnDestroy {
   constructor(private dfs: DynamicFormService) {}
 
   ngOnInit(): void {
-    this.form = this.dfs.createForm(this.template, this.vehicleTechRecord) as CustomFormGroup;
+    this.form = this.template ? (this.dfs.createForm(this.template, this.vehicleTechRecord) as CustomFormGroup) : ({} as CustomFormGroup);
```
Comment (tahayazici) [inline src/app/forms/custom-sections/dimensions/dimensions.component.ts:27]: Why is this check being applied now?

The new vehicle types do not use these custom sections, so they won't render on the page anyway.

Likewise, for the other 2 custom sections where you made the same change.

File: src/app/features/tech-record/components/tech-record-summary/tech-record-summary.component.ts [hunk 3]
```diff
@@ -111,9 +115,12 @@ export class TechRecordSummaryComponent implements OnInit {
         return this.getHgvTemplates();
       case VehicleTypes.TRL:
         return this.getTrlTemplates();
-      // TODO: Create light vehicle specific functions for this
```
Comment (aaron-downing) [inline src/app/features/tech-record/components/tech-record-summary/tech-record-summary.component.ts:114]: Is there anyway we can refactor this into a map

Comment (aaron-downing) [reply src/app/features/tech-record/components/tech-record-summary/tech-record-summary.component.ts:114]: You know I love maps

Comment (aaron-downing) [reply src/app/features/tech-record/components/tech-record-summary/tech-record-summary.component.ts:114]: Like why do we have a getter referencing getters, could we not have one getter that stores all of the template values in a map

**Review by tahayazici — APPROVED** [review]

File: src/app/features/tech-record/components/tech-record-summary/tech-record-summary.component.ts (hunk 3 above)
Comment (naathanbrown) [reply src/app/features/tech-record/components/tech-record-summary/tech-record-summary.component.ts:114]: fine I guess!!!

**Review by aaron-downing — APPROVED** [review]

**Review by leifkemp-bjss — APPROVED** [review]

**Review by tahayazici — APPROVED** [review]

**Comment by tahayazici** [comment]
Nice bit of refactoring. Well done!

**Review by Ian-Hodges — APPROVED** [review]

**Pull request merged**